core/src/operations/addu64.rs

```diff
@@ -0,0 +1,162 @@
+use core::borrow::Borrow;
+use core::borrow::BorrowMut;
+use p3_air::AirBuilder;
+use p3_field::Field;
+use sp1_derive::AlignedBorrow;
+use std::mem::size_of;
+
+use crate::air::SP1AirBuilder;
+use crate::air::Word;
+
+use crate::runtime::ExecutionRecord;
+use p3_field::AbstractField;
+
+/// A set of columns needed to compute the add of two words.
+#[derive(AlignedBorrow, Default, Debug, Clone, Copy)]
+#[repr(C)]
+pub struct AddOperationU64<T> {
+    /// The result of `a + b` in two 32bits limbs.
+    pub lo: Word<T>,
+    pub hi: Word<T>,
+
+    /// Trace.
+    pub carry: [T; 7],
+}
+
+impl<F: Field> AddOperationU64<F> {
+    pub fn populate(
+        &mut self,
+        record: &mut ExecutionRecord,
+        a_lo: u32,
```

Comment:
I think your life would be a lot better if this took in `u64, u64` and returned a u64? a later comment also points this out